### ISSUE TITLE
Default variable value not working inside the component, if that variable is not passed into the "locals" object

### ISSUE THREAD

**Issue by schirliuradu** [maizzle/framework#181]
- Maizzle Version: 1.0.8
- Node.js Version: 14.2.0

Hi Cosmin,

I was trying the new features that you've included in the last 2 weeks in the maizzle framework. I have to say that i love the new postHtml syntax compared to the nunkucks one. However, i would like to report some small things to you. 

Until the "0.9.1" maizzle version, using nunjucks, i defined some components that i used everytime in almost all emails, and i passed to each of them an object with all the configurations i wanted everytime i instantiated that component in a template. Inside of the component file, to each of those values contained in the object i gave a default value, with "old" nunjucks syntax (ex: ``` {{ data.textColor or '#ffffff' }} ```).

Here is an example of a simple cta component:

```
<table class="w-full text-center">
    <tr>
        <th>
            <table align="center" class="border-solid rounded mx-auto w-full" style="border-width: {{ data.borderWidth or '0' }}; border-color: {{ data.borderColor or '#ffffff' }}; background-color: {{ data.backgroundColor or '#ffa500' }};">
                <tr>
                    <th class="text-center uppercase" style="color: {{ data.textColor or '#000000' }}; mso-padding-alt: {{ data.paddingY or '15px' }} {{ data.paddingX or '30px' }};">
                        <a href="{{ data.ctaLink or '#' }}" class="no-underline font-bold" style="color: {{ data.textColor or '#ffffff' }}">
                            <span class="block" style="padding-left: {{ data.paddingX or '30px' }}; padding-right: {{ data.paddingX or '30px' }}; padding-top: {{ data.paddingY or '15px' }}; padding-bottom: {{ data.paddingY or '15px' }}; ">
                                {{ data.ctaText }}
                            </span>
                        </a>
                    </th>
                </tr>
            </table>
        </th>
    </tr>
</table>
```

Every time that i instantiated this component, i was able to pass just one parameter (the only one without the default value - ctaText, and other parameters if they changed, like colors and other things). 

I saw you did the same thing with the new postHtml components. In your example (vmlbg.html component) you do the same thing, with the new ``` || ``` or syntax.

```
<!--[if mso]>
<v:image src="{{ src || 'https://via.placeholder.com/600x400' }}" xmlns:v="urn:schemas-microsoft-com:vml" style="width:{{ width || 600 }}px;height:{{ height || 400 }}px;" />
<v:rect fill="false" stroke="false" style="position:absolute;width:{{ width || 600 }}px;height:{{ height || 400 }}px;">
<v:textbox inset="0,0,0,0"><div><![endif]-->
<content></content>
<!--[if mso]></div></v:textbox></v:rect><![endif]-->
```

 Well, the problem is that, if you don't pass in the "locals" object even just one of the variables used inside the component file, mmaizzle will not compile anything, throwing this error: 

```
✖ Failed to compile build_local/promotional.html
(node:6264) UnhandledPromiseRejectionWarning: ReferenceError: width is not defined
(Use `node --trace-warnings ...` to show where the warning was created)
(node:6264) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 3)
(node:6264) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

I think this is a must feature to have. If you have a big component, using 20 or 30 variables inside and just 3 or 5 or 10 changing everytime, you will not pass to that component 30 variables every time. 

Another thing i wanted to ask: while using the postHtml syntax ``` {{ ... }} ``` into the html files, vscode reports this syntax as errors, not everywhere, but just in the <style> tag and in the style="" attribute (i noticed just those 2 for the moment). Is there an extension for vscode that solves this? How do you handle this?

**Comment by cossssmin** [comment]
Hey Radu, thanks for the detalied feedback!

Using data inside components is now different with PostHTML - when you call `{{ src }}` inside a component, it tries to read it from the 'global' scope and it throws if it's undefined. So using `locals='{"src": "..."}'` adds `src` to that 'global' scope, which is why it will work with the attribute.

You can alternatively use the `page` object instead (which is always defined) and set variables on it, either in your config or through Front Matter, like in the [example](https://maizzle.com/docs/components/#example).

Basically, what Nunjucks did and PostHTML (currently) doesn't, is that it handled undefined variables differently, not throwing with a `ReferenceError` when something was undefined. I'll try to look into this with the PostHTML team, maybe we can improve this somehow... though I can't see how right now, since Node's [vm.runInContext](https://nodejs.org/api/vm.html#vm_vm_runincontext_code_contextifiedobject_options) is used.

VS Code syntax highlighting for `{{ }}` inside `<style>`: I don't know of any fix, I just live with it 🤷‍♂️
